symbolic-il2cpp/src/usym.rs

```diff
@@ -3,10 +3,104 @@
 //! This format can map il2cpp instruction addresses to managed file names and line numbers.
 
 use std::borrow::Cow;
+use std::error::Error;
+use std::fmt;
 use std::mem;
 use std::ptr;
 
-use anyhow::{Error, Result};
+use thiserror::Error;
+
+/// The error type for [`UsymError`].
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum UsymErrorKind {
+    /// Buffer to usym file is misaligned.
+    MisalignedBuffer,
+    /// The header to the usym file is missing or undersized.
+    BadHeader,
+    /// The magic string in the header is missing or malformed.
+    BadMagic,
+    /// The version string in the usym file's header is missing or malformed.
+    InvalidVersion,
```

Comment:
Nit: all other error cases where some datum is invalid are called `BadSomething`.